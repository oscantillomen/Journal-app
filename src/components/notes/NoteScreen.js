import React, { useEffect, useRef } from 'react'
import { NoteAppBar } from './NoteAppBar'
import { useSelector, useDispatch } from 'react-redux'
import { useForm } from '../../hooks/useForm';
import { activeNote, startDeleting } from '../../redux/actions/notes';

export const NoteScreen = () => {

    const { active: note } = useSelector(state => state.notes);
    const [ formValues, handleInputChange, reset ] = useForm(note);
    const { body, title, id } = formValues;
    const activeId = useRef(note.id);
    const dispatch = useDispatch();

    useEffect(() => {
        if( note.id !== activeId.current) {
            reset(note);
            activeId.current = note.id;
        }
    }, [note, reset]);

    useEffect(() => {
        dispatch(activeNote(formValues.id, { ...formValues }));
    }, [formValues, dispatch]);

    const handleDelete = () => {
        dispatch(startDeleting(id));
    }

    return (
        <div className="notes__main-content">
            <NoteAppBar />

            <div className="notes__content">
                <input 
                    type="text" 
                    placeholder="Some awesome title"
                    className="notes__title-input"
                    autoComplete="off"
                    name='title'
                    value={title}
                    onChange={handleInputChange}
                />

                <textarea 
                    className="notes__textarea"
                    placeholder="What happened today?"
                    name="body" 
                    id="" 
                    cols="30" 
                    rows="10"
                    value={body}
                    onChange={handleInputChange}
                ></textarea>

                { note.url &&
                    <div className="notes__image">
                        <img src={note.url} alt="note"/>
                    </div>
                }
            </div>
            <button 
                className="btn btn-danger"
                onClick={ handleDelete }
            >
                Delete
            </button>
        </div>
    )
}
