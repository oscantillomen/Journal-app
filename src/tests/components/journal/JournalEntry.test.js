import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import '@testing-library/jest-dom';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../redux/actions/notes';

const midlewares = [ thunk ];

const mockStore = configureStore(midlewares);

const initalState = {
	auth: {
		id: 'abc',
		name: 'Oscar Cantillo'
	},
	ui: {
		loading: false,
		msgError: null
	},
	notes: {
		active: null,
		notes: []
	}
};

let store = mockStore(initalState);
store.dispatch = jest.fn();

const note = {
	id: 10,
	date: 0,
	title: 'Hola',
	body: 'mundo',
	url: 'https://somewhere.com/picture.jpg'
};

const wrapper = mount(
	<Provider store={store}>
		<JournalEntry {...note} />
	</Provider>
);

describe('Pruebas en <JournalEntry/>', () => {
	test('Debe mostrarse correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe activar la note', () => {
		wrapper.find('.journal__entry').prop('onClick')();

		expect(store.dispatch).toHaveBeenCalledWith(activeNote(note.id, { ...note }));
	});
});
