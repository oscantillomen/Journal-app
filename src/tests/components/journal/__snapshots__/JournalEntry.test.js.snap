// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalEntry/> Debe mostrarse correctamente 1`] = `
<div
  className="journal__entry animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://somewhere.com/picture.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hola
    </p>
    <p
      className="journal__entry-content"
    >
      mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday 
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
